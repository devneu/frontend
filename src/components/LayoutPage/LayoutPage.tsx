import { Layout } from 'antd';
import React, { ReactElement } from 'react';

interface LayoutPageProps {
  content: ReactElement;
  sider: ReactElement;
}

const LayoutPage = ({ content, sider }: LayoutPageProps) => (
  <div className="post-page container">
    <Layout className="layout">
      <Layout.Content className="content">{content}</Layout.Content>
      <Layout.Sider className="sidebar" theme="light" width={400}>
        {sider}
      </Layout.Sider>
    </Layout>
  </div>
);

export default LayoutPage;
