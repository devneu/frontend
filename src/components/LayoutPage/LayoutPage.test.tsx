import React from 'react';
import renderer from 'react-test-renderer';
import LayoutPage from './LayoutPage';

describe('LayoutPage', () => {
  it('Should successfully render', () => {
    const wrapper = renderer.create(<LayoutPage content={<div>Content</div>} side={<div>Side</div>} container className="post-page" />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
