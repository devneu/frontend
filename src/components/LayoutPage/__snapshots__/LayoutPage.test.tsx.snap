// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutPage Should successfully render 1`] = `
<div
  className="post-page container"
>
  <section
    className="ant-layout layout"
  >
    <main
      className="ant-layout-content content"
    >
      <div>
        Content
      </div>
    </main>
    <aside
      className="ant-layout-sider ant-layout-sider-light sidebar"
      style={
        Object {
          "flex": "0 0 400px",
          "maxWidth": "400px",
          "minWidth": "400px",
          "width": "400px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      >
        <div>
          Side
        </div>
      </div>
    </aside>
  </section>
</div>
`;
